backend/app/core/social_service.py

```
@@ -1,297 +1,346 @@
-from sqlalchemy.orm import Session
-from sqlalchemy import and_, or_, func, text, desc, asc
-from typing import List, Optional, Dict, Any, Tuple
-from datetime import datetime, timedelta
 import logging
 import re
 from collections import Counter
+from datetime import datetime, timedelta
```

Ruff (0.11.9)

`datetime.timedelta` imported but unused

Remove unused import: `datetime.timedelta`

(F401)